ui/pages/confirmations/confirm-transaction/confirm-token-transaction-switch.js

```diff
@@ -144,14 +144,6 @@ export default function ConfirmTokenTransactionSwitch({ transaction }) {
             decimals={decimals}
             image={tokenImage}
             tokenAddress={tokenAddress}
-            onEdit={async ({ txData }) => {
```

Comment:
Actually wait, it looks like this component assumes that `onEdit` is always passed in

Reply:
Following the trail here: this is a non-required prop, passed into `confirm-token-transaction-base`. That component passes it to `confirm-transaction-base` (passed through the container untouched, to the component module).

It's called here: https://github.com/MetaMask/metamask-extension/blob/4b543cd1fc843547e10096a3591f28036aa303b9/ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js#L617

We try to call this in `handleEdit` without checking that it's defined.

Reply:
Oh but wait, we also set `showEdit` to `false` when `onEdit` is false, here: https://github.com/MetaMask/metamask-extension/blob/4b543cd1fc843547e10096a3591f28036aa303b9/ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js#L1099

Super confusing how this is setup, but never mind, this should work as-is.